test/telephone_number_test.rb

```diff
@@ -10,23 +14,109 @@ def test_valid_with_keys_returns_false
     assert TelephoneNumber.valid?("448444156790", "GB", [:shared_cost])
   end
 
-  def test_valid_without_keys_returns_true
-    assert TelephoneNumber.valid?("3175082489", "US")
-    assert TelephoneNumber.valid?("448444156790", "GB")
-  end
-
   def test_valid_with_invalid_country_returns_false
     refute TelephoneNumber.valid?("448444156790", "NOTREAL")
+    assert TelephoneNumber.invalid?("448444156790", "NOTREAL")
   end
 
-  def test_invalid_returns_false_if_valid
-    assert TelephoneNumber.valid?('6502530000', 'US')
-    refute TelephoneNumber.invalid?('6502530000', 'US')
+  def test_valid_types_is_not_empty_when_valid
+    @valid_numbers.each do |country, number_object|
+      number_object.each do |_name, number_data|
+        number_data.each do |_type, number|
+          assert TelephoneNumber.valid?(number, country)
+          refute TelephoneNumber.invalid?(number, country)
```

Comment:
Aren't these doing the same thing?

Reply:
They achieve the same result, but are calling two different methods in order to obtain that result. This test just ensures that I am always getting opposite results from those two methods. Does that make sense?

Reply:
Ahhh gotcha. Yea that makes sense.